crates/uv/src/commands/workspace/run.rs

```diff
@@ -58,53 +56,112 @@ pub(crate) async fn run(
         "python".to_string()
     };
 
-    // Copy the requirements into a set of overrides; we'll use this to prioritize
-    // requested requirements over those discovered in the project.
-    // We must retain these requirements as direct dependencies too, as overrides
-    // cannot be applied to transitive dependencies.
-    let overrides = requirements.clone();
+    // Discover and sync the workspace.
+    let workspace_env = if isolated {
+        None
+    } else {
+        debug!("Syncing workspace environment.");
 
-    if !isolated {
-        if let Some(workspace_requirements) = workspace::find_workspace()? {
-            requirements.extend(workspace_requirements);
-        }
-    }
+        let Some(workspace_requirements) = workspace::find_workspace()? else {
+            return Err(anyhow::anyhow!(
+                "Unable to find `pyproject.toml` for project workspace."
+            ));
+        };
 
-    // Detect the current Python interpreter.
-    // TODO(zanieb): Create ephemeral environments
-    // TODO(zanieb): Accept `--python`
-    let run_env = environment_for_run(
-        &requirements,
-        &overrides,
-        python.as_deref(),
-        isolated,
-        preview,
-        cache,
-        printer,
-    )
-    .await?;
-    let python_env = run_env.python;
+        let venv = PythonEnvironment::from_virtualenv(cache)?;
 
-    // Construct the command
-    let mut process = Command::new(&command);
-    process.args(&args);
+        // Sync the workspace requirements.
+        Some(sync_environment(venv, &workspace_requirements, preview, cache, printer).await?)
+    };
 
-    // Set up the PATH
-    debug!(
-        "Using Python {} environment at {}",
-        python_env.interpreter().python_version(),
-        python_env.python_executable().user_display().cyan()
-    );
-    let new_path = if let Some(path) = env::var_os("PATH") {
-        let python_env_path =
-            iter::once(python_env.scripts().to_path_buf()).chain(env::split_paths(&path));
-        env::join_paths(python_env_path)?
+    // If necessary, create an environment for the ephemeral requirements.
+    let tmpdir;
```

Comment:
That's curious, how is that implemented in rust, does this insert a conditional drop?